app/controllers/publish/courses/outcome_controller.rb

```diff
@@ -44,6 +49,52 @@ def current_step
       def errors
         params.dig(:course, :qualification) ? {} : { qualification: ['Select a qualification'] }
       end
+
+      def handle_qualification_update
+        if undergraduate_to_other_qualification?
+          redirect_to funding_type_publish_provider_recruitment_cycle_course_path(
+            @course.provider_code,
+            @course.recruitment_cycle_year,
+            @course.course_code,
+            previous_tda_course: true
+          )
+        else
+          Publish::Courses::AssignTdaAttributesService.new(@course, update: true).call if undergraduate_degree_with_qts?
```

Comment:
I'd prefer to move the update option out of the service. When I read assign I just expect  to change the attributes but no saving.

I'd recommend to do something like:

```ruby
if undergraduate_degree_with_qts?
 Publish::Courses::AssignTdaAttributesService.new(@course).call 
 @course.save
end
```

Reply:
Then on the other controller you don't need to pass the options as well.

Reply:
Good challenge and suggestion! Done here: https://github.com/DFE-Digital/publish-teacher-training/pull/4223/commits/1405dea92946b0f3e4e2e073a11a985ba9583f7f